src/resmom/linux/alps.c

```diff
@@ -3727,21 +3727,22 @@ inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
 						goto bad_vnl;
 					if (accel->data.gpu->memory) {
 						attr = "resources_available.accelerator_memory";
-						if (totseg == 0) {
+						if (totseg == 0 || vnode_per_numa_node) {
 							snprintf(utilBuffer,
 								sizeof(utilBuffer),
 								"%umb",
 								accel->data.gpu->memory);
-						} else if (vnode_per_numa_node) {
-							snprintf(utilBuffer,
-								sizeof(utilBuffer),
-								"@%s_%ld_0",
-								mpphost, node->node_id);
+							if (vnode_per_numa_node && !totseg) {
+								snprintf(utilBuffer,
+									 sizeof(utilBuffer),
+									 "@%s_%ld_0",
+									 mpphost, node->node_id);
+							}
+							if (vn_addvnr(nv, vname, attr,
+								      utilBuffer, 0,
+								      0, NULL) == -1)
+								goto bad_vnl;
```

Comment:
I think if we take a step back, we can see that the existing OSS code already handles the problem you are trying to solve.  Take a look: 
/*
3371	 				 * Avoid calling vn_addvnr() repeatedly when
3372	 				 * creating only one vnode per compute node.
3373	 				 */
3374					if (vnode_per_numa_node || (seg->ordinal == 0)) {
3375						if (vn_addvnr(nv, vname, attr, utilBuffer,
3376							0, 0, NULL) == -1)
3377							goto bad_vnl;
3378					}
You may need to modify the check on segment ordinal to match your new code, so you check for the right variables. The code above that (line 3662, 3670, 3730, and 3735) should go back to an if/else case without adding vnode_per_numa_node or totseg != 0 to the "if".  What do you think?